src/assets/js/cr-op/cropper-element.js

```diff
@@ -0,0 +1,130 @@
+import { setStyles } from './helpers';
+
+function addDragListener(container, dragItem, fn, key = '') {
+  let active = false;
+  let lastX = null;
+  let lastY = null;
+
+  function dragStart(event) {
+    if (event.target === dragItem || dragItem.contains(event.target)) {
+      event.stopImmediatePropagation();
+      active = true;
+    }
+  }
+
+  function dragEnd() {
+    lastX = null;
+    lastY = null;
+    active = false;
+  }
+
+  function drag(event) {
+    if (!active) {
+      return;
+    }
+
+    event.preventDefault();
+    let currentX;
+    let currentY;
+    if (event.type === 'touchmove') {
+      currentX = event.touches[0].clientX;
+      currentY = event.touches[0].clientY;
+    } else {
+      currentX = event.clientX;
+      currentY = event.clientY;
+    }
+
+    if (lastX != null && lastY != null) {
+      fn(currentX - lastX, currentY - lastY, key);
+    }
+
+    lastX = currentX;
+    lastY = currentY;
+  }
+
+  container.addEventListener('touchstart', dragStart);
+  container.addEventListener('touchend', dragEnd);
+  container.addEventListener('touchmove', drag);
+
+  container.addEventListener('mousedown', dragStart);
+  container.addEventListener('mouseup', dragEnd);
+  container.addEventListener('mousemove', drag);
+}
+
+const CROPPER_STYLE = {
+  position: 'absolute',
+  display: 'flex',
+  'flex-wrap': 'wrap',
+};
+const ONE_THIRD = `${100 / 3}%`;
+
+const CROPPER_BOX_STYLE = {
+  display: 'flex',
+  width: ONE_THIRD,
+  height: ONE_THIRD,
+  border: '1px solid rgba(255, 255, 255, .3)',
+  'box-sizing': 'border-box',
+  overflow: 'hidden',
+  cursor: 'move',
+};
+
+const BORDER = '2px solid white';
+
+const HANDLE_DIRECTIONS = {
+  1: 'nw',
+  2: 'n',
+  3: 'ne',
+  4: 'w',
+  6: 'e',
+  7: 'sw',
+  8: 's',
+  9: 'se',
+};
+
+const HANDLE_STYLES = [
+  i => i <= 3 && { 'border-top': BORDER },
+  i => i >= 7 && { 'border-bottom': BORDER, 'align-self': 'flex-end' },
+  i => i % 3 === 1 && { 'border-left': BORDER },
+  i => i % 3 === 0 && { 'border-right': BORDER, 'margin-left': 'auto' },
+  i => (i === 2 || i === 8) && { margin: '0 auto', cursor: 'ns-resize' },
+  i => (i === 4 || i === 6) && { 'align-self': 'center', cursor: 'ew-resize' },
+  i => (i === 1 || i === 9) && { cursor: 'nwse-resize' },
+  i => (i === 3 || i === 7) && { cursor: 'nesw-resize' },
+];
+
+export default function createCropperContainer(
+  container,
+  { onMove, onResize },
+) {
+  const cropperContainer = document.createElement('div');
+  setStyles(cropperContainer, CROPPER_STYLE);
+
+  for (let i = 1; i <= 9; i += 1) {
+    const box = document.createElement('div');
+    cropperContainer.appendChild(box);
+    setStyles(box, CROPPER_BOX_STYLE);
+    if (i > 3) {
+      setStyles(box, { 'border-top': 0 });
+    }
+    if (i % 3 !== 0) {
+      setStyles(box, { 'border-right': 0 });
+    }
+
+    if (i !== 5) {
+      const handle = document.createElement('div');
```

Comment:
should this be a `button`?

Reply:
Good question. I'm not sure. It's draggable, simply clicking it doesn't do anything. Is that still considered a button?

This also bring up the good point that this can not be used with the keyboard yet.

Reply:
I _think_ it's more a `button` than it is a `div`, but there's no (or little) end user benefit to it if it doesn't also work with keyboard. Would it be doable to connect arrow up / right / down /  left to whatever dragging in that direction does?

Reply:
Doable yes, though I think it'd take some time to get the UX right. I see two approaches right now:
1. what you said: every handle is focusable and the arrows keys do what dragging in that direction would've done
2. the whole thing is focusable, arrow keys move it, holding shift changes size in that direction

I think 2) is preferable, but I guess there'd be no way to communicate that interaction with accessibility attributes alone, would it?

Reply:
Agreed 2 is preferable. Maybe we can show a toast that explains to use shift if we see people use their keyboard?